api/proto/teleport/legacy/types/types.proto

```diff
@@ -5252,6 +5252,52 @@ message PluginListV1 {
   repeated PluginV1 plugins = 1;
 }
 
+// PluginStaticCredentialsV1 is a representation of static credentials for plugins.
+message PluginStaticCredentialsV1 {
+  option (gogoproto.goproto_stringer) = false;
+  option (gogoproto.stringer) = false;
+
+  // Header is the resource header for the plugin static credentials object.
+  ResourceHeader Header = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "",
+    (gogoproto.embed) = true
+  ];
+
+  // Spec contains the actual credentials for the object.
+  PluginStaticCredentialsSpecV1 Spec = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
```

Comment:
I don't think `jsonpb` uses the `jsontag` fields (these are only for Go's builtin JSON serializer). Could you verify what the serialized JSON looks like?

We had a similar conundrum introducing the `Plugin` type which also utilized `oneof`. We decided to use `snake_case` field names in `.proto`. The generator automatically transforms to `PascalCase` for Go fields, although it does not recognize names like `OAuth` and instead generates `Oauth`.

https://github.com/gravitational/teleport/pull/20990#pullrequestreview-1279025754 see @espadolini 's comments here.

Reply:
You're right, it doesn't respect these fields. I've gone ahead and removed most of the `gogo` extensions from this object.